tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecorator-specs.js

```diff
@@ -21,17 +23,76 @@ describe('ContextualPopupDecorator', function () {
 		});
 
 		describe('using 5-way', function () {
-			// The menu is open and the first item has focus
-			beforeEach(function () {
+			it('should focus second button on 5-way right', function () {
+				Page.spotlightRight();
+				expect(button2.self.isFocused()).to.be.true();
+			});
+		});
+	});
+
+	describe('using open', function () {
+		describe('using 5-way', function () {
+			it('should have Spotlight on button when ContextualPopup1 opens', function () {
+				let popupButton = $('#popupButton1');
+
 				Page.spotlightSelect();
+				expect(popupButton.isFocused()).to.be.true();
 			});
 
-			// [GT-28291] - 5-waySelectableActivator: Button Retains Spotlight when Popup Hides
-			it('should have Spotlight on close button when ContextualPopup opens', function () {
-				let popupButton = $('#popupButton');
+			it('should have Spotlight on button when ContextualPopup2 opens', function () {
+				let popupButton = $('#popupButton2');
 
+				button2.focus();
+				Page.spotlightSelect();
 				expect(popupButton.isFocused()).to.be.true();
 			});
+
+			it('should close the popup and should have Spotlight on button on auto dismiss the ContexualPopup2', function () {
+				let popupButton = $('#popupButton2');
+
+				button2.focus();
+				Page.spotlightSelect();
+				expect(popupButton.isFocused()).to.be.true();
+
+				Page.backKey();
+				expect(button2.isOpen).to.be.false();
+				expect(button2.self.isFocused()).to.be.true();
+			});
+
+			it('should not close the popup and should not have Spotlight on button on auto dismiss the ContexualPopup3', function () {
+				let popupButton = $('#popupButton3');
+
+				button3.focus();
+				Page.spotlightSelect();
+				expect(popupButton.isFocused()).to.be.true();
+
+				Page.backKey();
+				expect(button1.isOpen).to.be.true();
+				expect(button1.self.isFocused()).to.be.false();
+			});
+		});
+
+		describe('using pointer', function () {
+			it('should open when clicked', function () {
+				button1.self.click();
+				expect(button1.isOpen).to.be.true();
+			});
+
+			it('should close when clicking twice', function () {
+				button1.self.click();
+				button1.self.click();
+				expect(button1.isOpen).to.be.false();
+			});
+
+			it('should close when clicking outside', function () {
+				button1.self.click();
+
+				// Click in the area outside the ContextualPopupDecorator (in the empty space created by the wrapper)
+				const wrapper = $('.ThemeDecorator_ThemeDecorator_bg');
+				wrapper.click({x: 0, y: 0});
+
+				expect(button1.isOpen).to.be.false();
+			});
```

Comment:
Can you add a test for #button3 to show the contextualPopup is not closed when clicking outside?